compaction/compaction_manager.hh

```diff
@@ -63,6 +63,8 @@ inline owned_ranges_ptr make_owned_ranges_ptr(dht::token_range_vector&& ranges)
     return make_lw_shared<const dht::token_range_vector>(std::move(ranges));
 }
 
+future<owned_ranges_ptr> maybe_make_owned_ranges_ptr(locator::effective_replication_map_ptr erm, locator::host_id host_id);
```

Comment:
Why is this here, when the only user is in `replica/`?

Reply:
Can be defined there.
I can also move the definition and implementation of owned_ranges_ptr out of compaction.
It was there just because it was only used by cleanup, but it's not compaction specific really.